src/libertem/win_tweaks.py

```diff
@@ -27,11 +38,11 @@ def get_console_mode(stream=sys.stdin):
     getConsoleMode.argtypes, getConsoleMode.restype = ([wintypes.HANDLE, wintypes.LPDWORD],
                                                        wintypes.BOOL)
     mode = wintypes.DWORD(0)
-    if getConsoleMode(file_handle, byref(mode)):
-        return mode.value
-    else:
-        err = ctypes.get_last_error()
-        raise ctypes.WinError(err)
+    try:
+        if getConsoleMode(file_handle, byref(mode)):
+            return mode.value
+    except ctypes.WinError(ctypes.get_last_error()):
```

Comment:
I don't think this is right - why are you constructing an instance of the `WinError` class here in the `except` clause?

Reply:
Don't we have to supply `get_last_error()`?

Reply:
When **raising** a `WinError`, you have to provide the error code, yes. But if you are **handling** exceptions, this is different: only the **types** are matched in the `except` clause. The code that wants to **handle** the error can't know the details, like error codes or messages - they are only known *where the error occurs*!

Please, do read [the "Errors" chapter of the Python tutorial](https://docs.python.org/3/tutorial/errors.html), which explains the basics of exception handling, and let me know if this is now clear.